packages/@aws-cdk/aws-ecs/lib/base/base-task-definition.ts

```diff
@@ -0,0 +1,221 @@
+import iam = require('@aws-cdk/aws-iam');
+import cdk = require('@aws-cdk/cdk');
+import { ContainerDefinition, ContainerDefinitionProps } from '../container-definition';
+import { cloudformation } from '../ecs.generated';
+
+/**
+ * Basic task definition properties
+ */
+export interface BaseTaskDefinitionProps {
+  /**
+   * Namespace for task definition versions
+   *
+   * @default Automatically generated name
+   */
+  family?: string;
+
+  /**
+   * The IAM role assumed by the ECS agent.
+   *
+   * The role will be used to retrieve container images from ECR and
+   * create CloudWatch log groups.
+   *
+   * @default An execution role will be automatically created if you use ECR images in your task definition
+   */
+  executionRole?: iam.Role;
```

Comment:
Such a confusing name. In Lambda execution role is "taskRole" here. Are these the service semantics? Maybe we can diverge and call this `agentRole`?

Reply:
These are service semantics, yeah.

I think it's because of history; `executionRole` was first and `taskRole` was added later.

I'm not opposed to the change but let's bring in @SoManyHs for an opinion as well.

Reply:
Maybe `agentExecutionRole`?

Reply:
TaskRole was first, executionRole was later. The full name on the model is TaskExecutionRole -- I think it will be more confusing for those looking at the official docs if we call it anything else, even if it might collide with how other services name their roles.